components/operator-service/src/main/java/com/wso2telco/dep/operatorservice/dao/BlackListWhiteListDAO.java

```diff
@@ -53,55 +53,73 @@
 	/**
 	 * blacklist list given msisdns
 	 *
-	 * @param msisdns
-	 * @param apiID
-	 * @param apiName
-	 * @param userID
 	 * @throws Exception
 	 */
-	public void blacklist(MSISDNValidationDTO msisdns, final String apiID, final String apiName, final String userID)
+	public int blacklist(BlacklistWhitelistDTO dto)
 			throws Exception {
 
-		log.debug("BlackListWhiteListDAO.blacklist triggerd MSISDN[" + StringUtils.join(msisdns.getValidProcessed().toArray(), ",") + "] apiID:"
-				+ apiID + " apiName:" + apiName + " userID:" + userID);
-
 		StringBuilder sql = new StringBuilder();
-		sql.append(" INSERT INTO ");
-		sql.append(OparatorTable.BLACKLIST_MSISDN.getTObject());
-		sql.append("(PREFIX,MSISDN,API_ID,API_NAME,USER_ID,VALIDATION_REGEX)");
-		sql.append(" VALUES (?, ?, ?, ?, ?, ?)");
+		sql.append(" INSERT IGNORE INTO ");
+		sql.append(OparatorTable.API_BLACKLIST_WHITELIST.getTObject());
+            sql.append("(api_id,app_id,msisdn,action, SERVICE_PROVIDER, user)");
+		sql.append(" VALUES (?, ?, ?, ?,?, ?)");
 
 		Connection conn = null;
 		PreparedStatement ps = null;
 
+		int success = 0;
+
```

Comment:
Better to rethink and optimize this logic instead of iterating over and over again